app/src/main/java/com/hedvig/app/feature/offer/OfferActivity.kt

```diff
@@ -146,10 +146,14 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
     ) {
 
         premiumBoxTitle.setText(type.getStringId())
-        premium.text = BigDecimal(insuranceCost.monthlyNet.amount).toInt().toString()
-        if (BigDecimal(insuranceCost.monthlyDiscount.amount) > BigDecimal.ZERO) {
+        premium.text =
+            BigDecimal(insuranceCost.monthlyNet.fragments.monetaryAmountFragment.amount).toInt()
```

Comment:
`.fragments.monetaryAmountFragment.amount` Looks like this will be done in some places is it possible to create an extension for it? or cant we do it because the class is generated?

Reply:
I don't think we can generically make this type of lens function as the receiver is always of a different type and does not implement an interface 😞